src/main/java/com/nha/abdm/wrapper/hip/hrp/hipLink/HipLinkService.java

```diff
@@ -0,0 +1,207 @@
+/* (C) 2024 */
+package com.nha.abdm.wrapper.hip.hrp.hipLink;
+
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.nha.abdm.wrapper.common.ErrorResponse;
+import com.nha.abdm.wrapper.common.RequestManager;
+import com.nha.abdm.wrapper.common.Utils;
+import com.nha.abdm.wrapper.common.models.AckRequest;
+import com.nha.abdm.wrapper.common.models.VerifyOtp;
+import com.nha.abdm.wrapper.hip.hrp.database.mongo.repositories.LogsRepo;
+import com.nha.abdm.wrapper.hip.hrp.database.mongo.repositories.PatientRepo;
+import com.nha.abdm.wrapper.hip.hrp.database.mongo.services.PatientService;
+import com.nha.abdm.wrapper.hip.hrp.database.mongo.services.RequestLogService;
+import com.nha.abdm.wrapper.hip.hrp.database.mongo.tables.Patient;
+import com.nha.abdm.wrapper.hip.hrp.database.mongo.tables.RequestLog;
+import com.nha.abdm.wrapper.hip.hrp.discover.requests.OnDiscoverPatient;
+import com.nha.abdm.wrapper.hip.hrp.hipLink.requests.LinkAddCareContext;
+import com.nha.abdm.wrapper.hip.hrp.hipLink.requests.LinkAuthInit;
+import com.nha.abdm.wrapper.hip.hrp.hipLink.requests.LinkConfirm;
+import com.nha.abdm.wrapper.hip.hrp.hipLink.requests.helpers.*;
+import com.nha.abdm.wrapper.hip.hrp.hipLink.responses.LinkOnConfirmResponse;
+import com.nha.abdm.wrapper.hip.hrp.hipLink.responses.LinkOnInitResponse;
+import com.nha.abdm.wrapper.hip.hrp.hipLink.responses.LinkRecordsResponse;
+import java.util.Arrays;
+import java.util.UUID;
+import java.util.concurrent.TimeoutException;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import reactor.core.Exceptions;
+
+@Service
+public class HipLinkService implements HipLinkInterface {
+  @Autowired PatientRepo patientRepo;
+  @Autowired LogsRepo logsRepo;
+  @Autowired PatientService patientService;
+  @Autowired RequestManager requestManager;
+  @Autowired RequestLogService requestLogService;
+  private final String requesterType = "HIP";
+  private final String linkPurpose = "KYC_AND_LINK";
+
+  @Value("${linkAuthInitPath}")
+  public String linkAuthInitPath;
+
+  @Value("${linkConfirmAuthPath}")
+  public String linkConfirmAuthPath;
+
+  @Value("${linkAddContextsPath}")
+  public String linkAddContextsPath;
+
+  ResponseEntity<ObjectNode> responseEntity;
+  ErrorResponse errorResponse = new ErrorResponse();
+  private static final Logger log = LogManager.getLogger(HipLinkService.class);
+
+  public AckRequest hipAuthInit(LinkRecordsResponse linkRecordsResponse) {
+    try {
+      patientService.addPatient(linkRecordsResponse);
```

Comment:
This is not correct assumption. We do not add the patient in the beginning of hip auth init. I think what you meant here was to add care contexts. Even in that case, we should store in database only when it has been linked. Till that time it can stay in the database in request-log collection.

Reply:
Not adding the careContexts in the  initital stage, adding them only when we got status as success from ABDM